crates/uv-requirements/src/specification.rs

```diff
@@ -184,22 +184,26 @@ impl RequirementsSpecification {
                     ..Self::default()
                 }
             }
-            RequirementsSource::Pep723Script(path) => {
-                let script = match Pep723Script::read(&path).await {
-                    Ok(Some(script)) => Pep723Item::Script(script),
-                    Ok(None) => {
-                        return Err(anyhow::anyhow!(
-                            "`{}` does not contain inline script metadata",
-                            path.user_display(),
-                        ));
-                    }
-                    Err(Pep723Error::Io(err)) if err.kind() == std::io::ErrorKind::NotFound => {
-                        return Err(anyhow::anyhow!(
-                            "Failed to read `{}` (not found)",
-                            path.user_display(),
-                        ));
+            RequirementsSource::Pep723Script(source) => {
+                let script = if let Some(script) = source.script() {
+                    Pep723Item::Script(script.clone())
+                } else {
+                    match Pep723Script::read(source.path()).await {
+                        Ok(Some(script)) => Pep723Item::Script(script),
+                        Ok(None) => {
+                            return Err(anyhow::anyhow!(
+                                "`{}` does not contain inline script metadata",
+                                source.path().user_display(),
+                            ));
+                        }
+                        Err(Pep723Error::Io(err)) if err.kind() == std::io::ErrorKind::NotFound => {
```

Comment:
Haven't checked the rest properly yet, but the `if err.kind() == std::io::ErrorKind::NotFound` branches should be handled by fs_err already.